File: Intern.Models/PaginatedResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Intern.Models
{
    public class PaginatedResult<T>
    {
        public List<T> Data { get; set; }
        public int TotalCount { get; set; }
        public int PageNumber { get; set; }
        public int PageSize { get; set; }
        public bool Success { get; set; } = false;

        public int TotalPages => (int)Math.Ceiling((double)TotalCount / PageSize);
    }
}

File: ClientApp/src/app/models/paginated-result.ts
export class PaginatedResult<T> {
    data: Array<T>;
    totalCount: number;
    pageNumber: number;
    pageSize: number;

    constructor(data?: Array<T>, totalCount?: number, pageNumber?: number, pageSize?: number) {
        this.data = data ? data : [];
        this.totalCount = totalCount ? totalCount : 0;
        this.pageNumber = pageNumber ? pageNumber : 1;
        this.pageSize = pageSize ? pageSize : 10;
    }

    get totalPages(): number {
        return Math.ceil(this.totalCount / this.pageSize);
    }
}